### PR TITLE
Added support for client-pending-auth with WEB_AUTH

### PR DESCRIPTION
**Pull request by maurice2k**
Adds support for client-pending-auth with WEB_AUTH.

See https://github.com/Tunnelblick/Tunnelblick/issues/676.

### CODE REVIEW

**Comment by jkbullard** [comment]
Thanks!

Although this code ignores the flags, I assume it's useful for some situations, and dealing with the flags could be added later.

I think one change is needed: add a line with `return;` before the closing `}` at (new) line 4507. That will avoid dropping through to the existing code.

**Comment by maurice2k** [comment]
I've added the return statement and a comment that flags are currently not supported.
I had a look at the only other GUI I'm aware of that is supporting this (OpenVPN Connect) and they also just ignore the flags: https://github.com/OpenVPN/openvpn-gui/blob/7de26e65af2252d66114a0adadfd3136161c086b/openvpn.c#L1635

**Comment by jkbullard** [comment]
@maurice2k - Thanks again!

**Comment by schwabe** [comment]
That's nice. BUt I think to really use this Tunnelblick should add `--setenv IV_SSO webauth` to the commandline since otherwise you have to manually add that to configs.

**Comment by maurice2k** [comment]
> That's nice. BUt I think to really use this Tunnelblick should add `--setenv IV_SSO webauth` to the commandline since otherwise you have to manually add that to configs.

Good point, only had it in my config files. Added it in another PR: https://github.com/Tunnelblick/Tunnelblick/pull/767

**Comment by maurice2k** [comment]
> @maurice2k - Thanks again!

You're welcome and thanks for creating/maintaining TunnelblicK!

File: tunnelblick/VPNConnection.m [hunk 1]
```diff
@@ -4472,6 +4472,41 @@ -(void) processRealTimeOpenvpnOutputWithLine: (NSString *) line {
         [self sendStringToManagementSocket: @"hold release\r\n" encoding: NSASCIIStringEncoding];
         return;
     }
+
+    if (  [line hasPrefix: @">INFOMSG:WEB_AUTH:"]  ) {
+        // This is currently not supporting any flags; it just opens the given URL
+        // in the defauklt web browser.
```
Comment (schwabe) [inline tunnelblick/VPNConnection.m:4478]: Minor typo in default web browser.

**Comment by schwabe** [comment]
@maurice2k openvpn3-linux and OpenVPN for Android also support webauth method. OpeNVPN3-linux also ignores the flags. OpenVPN for Android actually implement an internal webview for the internal flag support and the javascript support from the spec for that but after OpenVPN Cloud/Connexa does not use that internal webview mode anymore, there is nobody left using that, so ignoring the flags should be fine for now.

File: tunnelblick/VPNConnection.m (hunk 1 above)
Comment (jkbullard) [reply tunnelblick/VPNConnection.m:4478]: Thanks, @schwabe; I've fixed that

**Comment by jkbullard** [comment]
Thanks for all the help, @schwabe.

**Pull request merged**